#!/usr/bin/env python3
<?xml version="1.0"?>
<launch>
  <!-- Define arguments that the included launch file expects -->
  <arg name="config" default="true"/>
  <arg name="rviz_config" default="$(find sawyer_moveit_config)/launch/moveit.rviz" />
  <arg name="db" default="false" />
  <arg name="debug" default="false" />
  <arg name="load_robot_description" default="false"/>
  <arg name="controller_box" default="true"/>
  <arg name="electric_gripper" default="false"/>
  <arg name="kinect" default="false" />
  <arg name="xtion" default="false" />
  <arg name="camera_link_pose" default="0.15 0.075 0.5 0.0 0.7854 0.0"/>

  <!-- Include the original Sawyer MoveIt launch file -->
  <include file="$(find sawyer_moveit_config)/launch/sawyer_moveit.launch">
    <arg name="config" value="$(arg config)"/>
    <arg name="rviz_config" value="$(arg rviz_config)"/>
    <arg name="db" value="$(arg db)"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="load_robot_description" value="$(arg load_robot_description)"/>
    <arg name="controller_box" value="$(arg controller_box)"/>
    <arg name="electric_gripper" value="$(arg electric_gripper)"/>
    <arg name="kinect" value="$(arg kinect)"/>
    <arg name="xtion" value="$(arg xtion)"/>
    <arg name="camera_link_pose" value="$(arg camera_link_pose)"/>
  </include>

  <!-- Wait for move_group to be ready, then load the pillar scene -->
  <node name="load_pillar_scene" pkg="moveit_ros_planning_interface" type="moveit_publish_scene_from_text" output="screen">
    <param name="scene_text" value="$(find sawyer_moveit_config)/sawyer_scenes/sawyer_pillar.scene"/>
    <remap from="planning_scene" to="/move_group/monitored_planning_scene"/>
  </node>

</launch> 